application/src/main/java/org/thingsboard/server/service/cf/ctx/state/SingleValueArgumentEntry.java

```diff
@@ -90,7 +92,14 @@ public Object getValue() {
 
     @Override
     public TbelCfArg toTbelCfArg() {
-        return new TbelCfSingleValueArg(ts, kvEntryValue.getValue());
+        Object value;
+        try {
+            value = JacksonUtil.readValue(kvEntryValue.getValueAsString(), new TypeReference<>() {
```

Comment:
You try to parse JSON for ALL types of entries? Can't you check that kvEntryValue is instance of JsonDataEntry?